morpheus/pipeline/pipeline.py

```diff
@@ -397,6 +398,13 @@ def stop(self):
         Stops all running stages and the underlying MRC pipeline.
         """
 
+        # Only execute this once
```

Comment:
Is it possible to have a race here? Do we need a mutex?

Reply:
seems unlikely but possible. i can add a mutex to be safe. do you think i should also add mutex for build and start? cc: @mdemoret-nv